### PR TITLE
22 feature section

### LINKED ISSUE #37: Features component
Image

### PR DESCRIPTION
**Pull request by sguming**
image
[acf-export-2024-07-06.json](https://github.com/user-attachments/files/16116233/acf-export-2024-07-06.json)
11am-5pm thursday Friday and Saturday
19 hours

### CODE REVIEW

**Comment by sguming** [comment]
Conflict solve

File: html/wp-content/themes/ailurotech/src/scss/feature-section.scss [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    width: 0;
+    background-color: rgba(51, 102, 255, 1);
+    border-radius: 10px 0 0 10px; 
+}
+
+
+@media  (max-width: 600px) {
+    .feature-title {
+        width: 100%;
+        font-size: 75%; /* 字体大小为 3vw 或者 20px 中的较小值 */
```
Comment (RuntaoZhuge) [inline html/wp-content/themes/ailurotech/src/scss/feature-section.scss]: Remove chatgpt comments please

File: html/wp-content/themes/ailurotech/index.php [hunk 2]
```diff
@@ -48,6 +48,25 @@
...
+$feature_title=get_field('feature_title');
+
+$feature_left_title=get_field('feature_left_title');
+$feature_left_details=get_field('feature_left_details');
+$feature_right_details_title=get_field('feature_right_details_title');
+$feature_right_details_number=get_field('feature_right_details_number');
+
+$context['feature_title']=get_field('feature_title');
+$context['feature_right_title']=get_field('feature_right_title');
+$context['feature_icon'] = explode(" ", $feature_icon);
```
Comment (RuntaoZhuge) [inline html/wp-content/themes/ailurotech/index.php]: Can you make these consistent? This is very confusing when sometimes sepreated by space, !, and ,

File: html/wp-content/themes/ailurotech/index.php [hunk 3]
```diff
@@ -48,6 +48,25 @@
...
+$context['feature_left_title']=explode(",",$feature_left_title);
+$context['feature_left_details']=explode("!",$feature_left_details);
+$context['feature_right_details_title']=explode(",",$feature_right_details_title);
+$context['feature_right_details_number']=explode(" ",$feature_right_details_number);
```
Comment (RuntaoZhuge) [inline html/wp-content/themes/ailurotech/index.php]: Then you can make this more concise by just passing in a string and having the out put as a list

File: html/wp-content/themes/ailurotech/src/scss/_footer.scss [hunk 4]
```diff
@@ -29,6 +29,10 @@
     margin-bottom: 40px;
 }
 
+#footer{
+    margin-top: 80px;
```
Comment (RuntaoZhuge) [inline html/wp-content/themes/ailurotech/src/scss/_footer.scss]: This looks kind of hard coded need more testing

File: html/wp-content/themes/ailurotech/views/partial/feature-section.twig [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<div id="feature" class="feature-container">
+  <div class="feature-title">
+    <h1>{{feature_title}}</h1>
+  </div>
+  <div class="feature-left">
+    {% for index in feature_left_title %}
+      <div class="feature-left-container">
+        <img class="feature-left-icon" src="{{feature_icon[loop.index-1]}}" alt="feature icon" />
+        <p><span class="feature-left-title">{{feature_left_title[loop.index-1]}}</span> <span class="feature-left-details">{{feature_left_details[loop.index-1]}}</span></p>
+      </div>
```
Comment (RuntaoZhuge) [inline html/wp-content/themes/ailurotech/views/partial/feature-section.twig]: What would happen if I only have 1 item in the loop?[loop.index-1] would it be pointing to -1 index?

File: html/wp-content/themes/ailurotech/views/base.twig [hunk 6]
```diff
@@ -16,6 +16,7 @@
     <main>
       {% block content %}
       {% endblock %}
+      {% include 'partial/feature-section.twig' %}
```
Comment (RuntaoZhuge) [inline html/wp-content/themes/ailurotech/views/base.twig]: Move this into content don't place directly in base

**Review by RuntaoZhuge — CHANGES_REQUESTED** [review]
@codeWorld666 Can you have a functional testing on this ticket?

**Comment by codeWorld666** [comment]
1. Please remove the debug notice at the top of the page.
2. please remove the extra percentage symbols
3. no blue progress bar
image

image

image
@RuntaoZhuge @sguming

File: html/wp-content/themes/ailurotech/views/partial/index.twig [hunk 7]
```diff
@@ -1,6 +1,8 @@
 {% extends 'base.twig' %}
 
 {% block content %}
-	{% include 'partial/welcome-section.twig' %}
+        {% include 'partial/welcome-section.twig' %}
 	{% include 'partial/contact-section.twig' %}
+        {% include 'partial/feature-section.twig' %}
```
Comment (codeWorld666) [inline html/wp-content/themes/ailurotech/views/partial/index.twig:5]: Please move the feature section above the contact section.

**Comment by sguming** [comment]
[acf-export-2024-07-15.json](https://github.com/user-attachments/files/16230847/acf-export-2024-07-15.json)

**Comment by sguming** [comment]
1,  all of acf get_field in this component i change to use "!", and please modify them, or just use wp-all-import plugin to import 
[Pages-Export-2024-July-15-2117.csv](https://github.com/user-attachments/files/16240518/Pages-Export-2024-July-15-2117.csv) and If have any further issues, please contact me.
2, i have done the null checking of all the component i finished

**Review by RuntaoZhuge — APPROVED** [review]

**Pull request merged**